modules/core/src/commonMain/kotlin/fr/acinq/lightning/crypto/RouteBlinding.kt

```diff
@@ -117,19 +122,23 @@ object RouteBlinding {
         encryptedPayload: ByteVector
     ): Either<InvalidTlvPayload, Pair<ByteVector, PublicKey>> {
         val sharedSecret = Sphinx.computeSharedSecret(pathKey, privateKey)
-        val rho = Sphinx.generateKey("rho", sharedSecret)
-        return try {
-            val decrypted = ChaCha20Poly1305.decrypt(
-                rho.toByteArray(),
-                Sphinx.zeroes(12),
-                encryptedPayload.dropRight(16).toByteArray(),
-                byteArrayOf(),
-                encryptedPayload.takeRight(16).toByteArray()
-            )
-            val nextPathKey = Sphinx.blind(pathKey, Sphinx.computeBlindingFactor(pathKey, sharedSecret))
-            Either.Right(Pair(ByteVector(decrypted), nextPathKey))
-        } catch (_: Throwable) {
-            Either.Left(CannotDecodeTlv(OnionPaymentPayloadTlv.EncryptedRecipientData.tag))
+        val nextPathKey = Sphinx.blind(pathKey, Sphinx.computeBlindingFactor(pathKey, sharedSecret))
+        if (encryptedPayload.isEmpty()) {
```

Comment:
The compact format should still be disallowed for payments and for standard offers. We should enable it explicitly only when we need it.